relayer/crates/starknet-chain-components/src/impls/queries/packet_is_cleared.rs

```diff
@@ -0,0 +1,25 @@
+use cgp::prelude::*;
+use hermes_chain_components::traits::queries::packet_is_cleared::PacketIsClearedQuerier;
+use hermes_chain_components::traits::types::ibc::{
+    HasChannelIdType, HasPortIdType, HasSequenceType,
+};
+use hermes_cosmos_chain_components::components::client::PacketIsClearedQuerierComponent;
+
+#[new_cgp_provider(PacketIsClearedQuerierComponent)]
+impl<Chain, Counterparty> PacketIsClearedQuerier<Chain, Counterparty>
+    for QueryStarknetPacketIsCleared
+where
+    Chain: HasChannelIdType<Counterparty>
+        + HasPortIdType<Counterparty>
+        + HasSequenceType<Counterparty>
+        + HasAsyncErrorType,
+{
+    async fn query_packet_is_cleared(
+        _chain: &Chain,
+        _port_id: &Chain::PortId,
+        _channel_id: &Chain::ChannelId,
+        _sequence: &Chain::Sequence,
+    ) -> Result<bool, Chain::Error> {
+        Ok(false) // stub
```

Comment:
why is this not implemented?

Reply:
I just haven't figured out how to implement this. It seems like we cannot use the `packet_commitment` call from the Cairo contract, as it would error if the commitment entry doesn't exist. But we cannot distinguish errors caused by the packet commitment not found, or by other reasons.

We probably need a separate endpoint to be able to query whether a packet commitment is found, or we could modify the existing endpoint to return an `Option`. @Farhad-Shabani how do you think?

In any case, this PR is mainly to scaffold the boilerplate for the new packet clearing, so we can fill in the proper implementation in another PR.

Reply:
Ah I see.

Since this is specific to this implementation, let's wrap this PR and move this discussion to a new issue.

- Please open an issue for this.
- Please add a comment here with a brief description with the issue number.

Then you can merge this onto `main` 🙌🏼

Reply:
Filed https://github.com/informalsystems/ibc-starknet/issues/291

Reply:
Opened https://github.com/informalsystems/ibc-starknet/pull/297